asset-transfer-events/application-gateway-typescript/src/app.ts

```diff
@@ -0,0 +1,363 @@
+/**
+ * Submit a transaction synchronously to delete an asset by ID.
+ */
+async function deleteAssetByID(contract:Contract): Promise<void>{
+    console.log('\n--> Submit Transaction: DeleteAsset asset70');
+
+    await contract.submitTransaction(
+        'DeleteAsset',
+        assetId
+    );
+
+    console.log('*** Transaction committed successfully');
+}
+
+/**
+ * submitTransaction() will throw an error containing details of any error responses from the smart contract.
+ */
+async function updateNonExistentAsset(contract: Contract): Promise<void>{
+    console.log('\n--> Submit Transaction: UpdateAsset asset70, asset70 does not exist and should return an error');
+
+    try {
+        await contract.submitTransaction(
+            'UpdateAsset',
+            'asset70',
+            'blue',
+            '5',
+            'Tomoko',
+            '300',
+        );
+        console.log('******** FAILED to return an error');
+    } catch (error) {
+        console.log('*** Successfully caught the error: \n', error);
+    }
+}
+
+/**
+ * Verify asset details.
+ */
+function checkAsset(mspId:string, asset:{
+    ID: string,
+    Color: string,
+    Size: string,
+    Owner: string,
+    AppraisedValue: string,
+    asset_properties:{
+        object_type: string,
+		asset_id: string,
+		Price: string,
+		salt: string
+    }
+}, color:string, size:string, owner:string, appraisedValue:string, price?:string) {
+    console.log(`<-- Query results from ${mspId}`);
+
+    console.log(`*** verify asset ${asset.ID}`);
+
+    if (asset) {
+        if (asset.Color === color) {
+            console.log(`*** asset ${asset.ID} has color ${asset.Color}`);
+        } else {
+            console.log(`*** asset ${asset.ID} has color of ${asset.Color}`);
+        }
+        if (asset.Size === size) {
+            console.log(`*** asset ${asset.ID} has size ${asset.Size}`);
+        } else {
+            console.log(`*** Failed size check from ${mspId} - asset ${asset.ID} has size of ${asset.Size}`);
+        }
+        if (asset.Owner === owner) {
+            console.log(`*** asset ${asset.ID} owned by ${asset.Owner}`);
+        } else {
+            console.log(`*** Failed owner check from ${mspId} - asset ${asset.ID} owned by ${asset.Owner}`);
+        }
+        if (asset.AppraisedValue === appraisedValue) {
+            console.log(`*** asset ${asset.ID} has appraised value ${asset.AppraisedValue}`);
+        } else {
+            console.log(`*** Failed appraised value check from ${mspId} - asset ${asset.ID} has appraised value of ${asset.AppraisedValue}`);
+        }
+        if (price) {
+            if (asset.asset_properties && asset.asset_properties.Price === price) {
+                console.log(`*** asset ${asset.ID} has price ${asset.asset_properties.Price}`);
+            } else {
+                console.log(`*** Failed price check from ${mspId} - asset ${asset.ID} has price of ${asset.asset_properties.Price}`);
+            }
+        }
+    }
+}
```

Comment:
This is looking more like a roll-your-own unit test and adds a huge amount of distracting clutter to the sample. It's nothing to do with eventing so I would remove it.